Bug Fix for `phasor_calibrate` when skipping axes with harmonic axis

bruno-pannunzio opened the pull request:
## Description
This PR proposes a fix for a bug when using the `skip_axis` parameter in `phasor_calibrate` function in the `phasor` module, when a harmonic axis is present.

## Checklist

- [x] The pull request title and description are concise.
- [ ] Related issues are linked in the description.
- [ ] New dependencies are explained.
- [x] The source code and documentation can be distributed under the [MIT license](https://www.phasorpy.org/docs/stable/license/).
- [x] The source code adheres to [code standards](https://www.phasorpy.org/docs/stable/contributing/#code-standards).
- [x] New classes, functions, and features are thoroughly [tested](https://www.phasorpy.org/docs/stable/contributing/#tests).
- [x] New, user-facing classes, functions, and features are [documented](https://www.phasorpy.org/docs/stable/contributing/#documentation).
- [ ] New features are covered in tutorials.
- [x] No files other than source code, documentation, and project settings are added to the repository.

bruno-pannunzio commented:
Hi @cgohlke! I came across with (another) bug in `phasor_calibrate` when working on the wavelet filter implementation.

The issue occurs when using the `skip_axis` parameter in `phasor_calibrate` with multiple harmonics or frequencies. The parameter was not working because the known theoretical phasor coordinates for the fluorophore of known lifetime were not broadcasted to the skipped axis to match the shape of the measured phasor coordinates.

There was a test that was meant to catch this, but in during the implementation I forgot to modify it and it resulted in the same test as without the `skip_axis` parameter with multiple harmonics. I fixed this and also the test with multiple frequencies instead of multiple harmonics.

cgohlke commented:
Hi @bruno-pannunzio Can you post a simple example that currently fails and passes with this PR? I got lost in the parametrized tests.

bruno-pannunzio commented:
Sure Christoph, I should have open an issue first describing the problem. But here is an example where I added a dimension to data to test this and it raises an error with the current state of the main branch, but not with this PR:

```Python
import numpy

from phasorpy.datasets import fetch
from phasorpy.io import signal_from_imspector_tiff
from phasorpy.phasor import (
    phasor_calibrate,
    phasor_from_signal,
)

def add_axis_and_repeat(array, axis=1, repeats=2):
    array_expanded = numpy.expand_dims(array, axis=axis)
    return numpy.repeat(array_expanded, repeats=repeats, axis=axis)

harmonic = [1, 2, 3]

signal = signal_from_imspector_tiff(fetch('Embryo.tif'))
calibration_signal = signal_from_imspector_tiff(
    fetch('Fluorescein_Embryo.tif')
)

mean, real, imag = phasor_from_signal(signal, axis=0, harmonic=harmonic)
calibration_mean, calibration_real, calibration_imag = phasor_from_signal(
    calibration_signal, axis=0, harmonic=harmonic
)

mean = add_axis_and_repeat(mean, axis=0)
real = add_axis_and_repeat(real)
imag = add_axis_and_repeat(imag)
calibration_mean = add_axis_and_repeat(calibration_mean, axis=0)
calibration_real = add_axis_and_repeat(calibration_real)
calibration_imag = add_axis_and_repeat(calibration_imag)

real, imag = phasor_calibrate(
    real,
    imag,
    calibration_mean,
    calibration_real,
    calibration_imag,
    frequency=80,
    lifetime=4.2,
    harmonic=harmonic,
    skip_axis=0,
)
```

The idea is to test skipping the axis I added, but it raises this error:

```
ValueError: operands could not be broadcast together with shapes (3,2) (3,2) (3,) (3,)
````
